server/routes/routes.js

```diff
@@ -236,22 +235,108 @@ async function getConnectedStops(stop){
 
 }
 
-async function findRoutes(startStops, endStops, currentRoutes){
+async function findRoutes(startStops, endStops, currentRoutes, directTrips){
+  let maxTransfers = 2
+  while (maxTransfers > 0) {
     for (const stop of endStops){
       const connectedStops = await getConnectedStops(stop)
       for (const cstop of connectedStops){
         if (startStops.some(sstop => sstop.stopId === cstop.stopId)){
-          const route =  await findDirectRoutes([cstop], connectedStops)
-          route.push(await findDirectRoutes(connectedStops, endStops))
+          const route =  await findDirectRoutes([cstop], connectedStops, pastTripIds = directTrips)
+          route.push(await findDirectRoutes(connectedStops, endStops, pastTripIds = directTrips))
           currentRoutes.push(route)
         }
       }
     }
-    return currentRoutes
+    maxTransfers --
+  }
+  return currentRoutes
+}
+
+
+// This function can definitely be cleaner, but for the purpose of unblocking myself, I leaving it for now.
```

Comment:
Nice! Yeah, like you mentioned, I also see some places where we could have early returns to reduce nesting.